app/main.py

```diff
@@ -1,2 +1,96 @@
+class Node:
+    def __init__(self, key: int, value: int) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key)
+        self.next = None
+
+
 class Dictionary:
-    pass
+    def __init__(self, initial_capacity: int = 8,
+                 load_factor: float = 0.75) -> None:
+        self.capacity = initial_capacity
+        self.load_factor = load_factor
+        self.size = 0
+        self.hash_table = [None] * self.capacity
+
+    def __len__(self) -> int:
+        return self.size
+
+    def _hash(self, key: int) -> int:
+        return hash(key) % self.capacity
+
+    def _resize(self) -> None:
+        old_table = self.hash_table
+        self.capacity *= 2
+        self.hash_table = [None] * self.capacity
+        self.size = 0
+
+        for node in old_table:
+            while node:
+                self.__setitem__(node.key, node.value)
+                node = node.next
+
+    def __setitem__(self, key: int, value: int) -> None:
```

Comment:
```suggestion
    def __setitem__(self, key: Hashable, value: Any) -> None:
```
please fix everywhere